tests/integration/test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py

```diff
@@ -0,0 +1,129 @@
+pytestmark = [pytest.mark.server, pytest.mark.tier(level=1)]
+
+# Cases metadata and its ids.
+cases_path = Path(TEST_CASES_PATH, 'cases_multi_agent_protocols_communication.yaml')
+config_path = Path(CONFIGS_PATH, 'config_multi_agent_protocols_communication.yaml')
+test_configuration, test_metadata, cases_ids = get_test_cases_data(cases_path)
+test_configuration = load_configuration_template(config_path, test_configuration, test_metadata)
+
+daemons_handler_configuration = {'all_daemons': True}
+
+local_internal_options = {REMOTED_DEBUG: '2'}
+
+injectors = []
+
+
+def send_event(event, protocol, manager_port, agent):
+    """Send an event to the manager"""
+
+    sender, injector = connect(agent, manager_port = manager_port, protocol = protocol)
+    injectors.append(injector)
+    sender.send_event(event)
+    return injector
+
+# Test function.
+@pytest.mark.parametrize('test_configuration, test_metadata',  zip(test_configuration, test_metadata), ids=cases_ids)
+def test_multi_agent_protocols_communication(test_configuration, test_metadata, configure_local_internal_options, truncate_monitored_files,
+                            set_wazuh_configuration, daemons_handler, simulate_agents):
+
+    '''
+    description: Check agent-manager communication with several agents simultaneously via TCP, UDP or both.
+                 For this purpose, the test will create all the agents and select the protocol using Round-Robin. Then,
+                 an event and a message will be created for each agent created. Finally, it will search for
+                 those events within the messages sent to the manager.
+
+
+    parameters:
+        - test_configuration
+            type: dict
+            brief: Configuration applied to ossec.conf.
+        - test_metadata:
+            type: dict
+            brief: Test case metadata.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Configure the Wazuh local internal options using the values from `local_internal_options`.
+        - daemons_handler:
+            type: fixture
+            brief: Restart service once the test finishes stops the daemons.
+        - simulate_agents
+            type: fixture
+            brief: create agents
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Apply changes to the ossec.conf configuration.
+    '''
+    agents = simulate_agents
+    senders = []
+
+
+    manager_port = test_metadata['port']
+    protocol = test_metadata['protocol']
+    search_patterns = []
+    send_event_threads = []
+
+    # Read the events log data
+    log_monitor_archives = FileMonitor(ARCHIVES_LOG_PATH)
+
+    for agent in agents:
+
+        # Generate custom events for each agent
+        search_pattern = f"test message from agent {agent.id}"
```

Comment:
Can these strings be defines?

Reply:
It is not necessary because it is part of the test.